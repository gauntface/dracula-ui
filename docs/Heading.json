{
  "description": "Heading is used to display headlines and other forms of hierarchical Text.\n\nHeadings are similar to the base Text component, but restricted to certain\nsizes and font weight.",
  "displayName": "Heading",
  "methods": [],
  "props": {
    "size": {
      "defaultValue": null,
      "description": "The size and type of Heading to be used.\nDenotes hierarchy.",
      "name": "size",
      "parent": {
        "fileName": "src/components/Typography/Heading.tsx",
        "name": "HeadingProps"
      },
      "required": false,
      "type": {
        "name": "\"heading-1\" | \"heading-2\" | \"heading-3\" | \"heading-4\" | undefined"
      }
    },
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI color to be applied to the Heading.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Typography/Heading.tsx",
        "name": "HeadingProps"
      },
      "required": false,
      "type": {
        "name": "\"black\" | \"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Controls the spacing between the Heading component and its parent and siblings.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Typography/Heading.tsx",
        "name": "HeadingProps"
      },
      "required": false,
      "type": {
        "name": "\"large\" | \"none\" | \"noneX\" | \"noneY\" | \"small\" | \"smallX\" | \"smallY\" | \"medium\" | \"mediumX\" | \"mediumY\" | \"largeX\" | \"largeY\" | (\"large\" | \"none\" | \"noneX\" | \"noneY\" | \"small\" | ... 6 more ... | \"largeY\")[] | undefined"
      }
    }
  }
}
