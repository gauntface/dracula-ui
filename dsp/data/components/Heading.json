{
  "class": "component",
  "type": "page",
  "id": "Heading",
  "name": "Heading",
  "last_updated": "2021-03-15T05:45:21.419Z",
  "last_updated_by": "System",
  "description": "Heading is used to display headlines and other forms of hierarchical Text.\n\nHeadings are similar to the base Text component, but restricted to certain\nsizes and font weight.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "heading",
    "languages": {
      "javascriptreact": "<Heading>Heading</Heading>;\n",
      "html": "<h1 class=\"drac-heading drac-heading-1 drac-text-white\">Heading</h1>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<h1 class=\"drac-heading drac-heading-1 drac-text-white\">Heading</h1>\n",
      "react": "<Heading>Heading</Heading>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Heading.png",
      "svgData": "./dsp/assets/svgs/Heading.svg"
    },
    {
      "title": "Sizes",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"white\" class=\"drac-heading drac-heading-1 drac-text-white\">\n      heading-1\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h2 color=\"white\" class=\"drac-heading drac-heading-2 drac-text-white\">\n      heading-2\n    </h2>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h3 color=\"white\" class=\"drac-heading drac-heading-3 drac-text-white\">\n      heading-3\n    </h3>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h4 color=\"white\" class=\"drac-heading drac-heading-4 drac-text-white\">\n      heading-4\n    </h4>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h5 color=\"white\" class=\"drac-heading drac-heading-5 drac-text-white\">\n      heading-5\n    </h5>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h6 color=\"white\" class=\"drac-heading drac-heading-6 drac-text-white\">\n      heading-6\n    </h6>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-1\">\n      heading-1\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-2\">\n      heading-2\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-3\">\n      heading-3\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-4\">\n      heading-4\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-5\">\n      heading-5\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-6\">\n      heading-6\n    </Heading>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/HeadingSizes.png",
      "svgData": "./dsp/assets/svgs/HeadingSizes.svg"
    },
    {
      "title": "Colors",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"white\" class=\"drac-heading drac-heading-1 drac-text-white\">\n      white\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"black\" class=\"drac-heading drac-heading-1 drac-text-black\">\n      black\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"blackSecondary\"\n      class=\"drac-heading drac-heading-1 drac-text-black-secondary\"\n    >\n      blackSecondary\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"blackLight\"\n      class=\"drac-heading drac-heading-1 drac-text-black-light\"\n    >\n      blackLight\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"cyan\" class=\"drac-heading drac-heading-1 drac-text-cyan\">\n      cyan\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"green\" class=\"drac-heading drac-heading-1 drac-text-green\">\n      green\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"orange\" class=\"drac-heading drac-heading-1 drac-text-orange\">\n      orange\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"pink\" class=\"drac-heading drac-heading-1 drac-text-pink\">\n      pink\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"purple\" class=\"drac-heading drac-heading-1 drac-text-purple\">\n      purple\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"red\" class=\"drac-heading drac-heading-1 drac-text-red\">red</h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"yellow\" class=\"drac-heading drac-heading-1 drac-text-yellow\">\n      yellow\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"purpleCyan\"\n      class=\"drac-heading drac-heading-1 drac-text-purple-cyan\"\n    >\n      purpleCyan\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"yellowPink\"\n      class=\"drac-heading drac-heading-1 drac-text-yellow-pink\"\n    >\n      yellowPink\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"cyanGreen\"\n      class=\"drac-heading drac-heading-1 drac-text-cyan-green\"\n    >\n      cyanGreen\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"pinkPurple\"\n      class=\"drac-heading drac-heading-1 drac-text-pink-purple\"\n    >\n      pinkPurple\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1 color=\"animated\" class=\"drac-heading drac-heading-1 drac-text-animated\">\n      animated\n    </h1>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\">white</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"black\">black</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"blackSecondary\">blackSecondary</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"blackLight\">blackLight</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"cyan\">cyan</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"green\">green</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"orange\">orange</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"pink\">pink</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"purple\">purple</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"red\">red</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"yellow\">yellow</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"purpleCyan\">purpleCyan</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"yellowPink\">yellowPink</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"cyanGreen\">cyanGreen</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"pinkPurple\">pinkPurple</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"animated\">animated</Heading>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/HeadingColors.png",
      "svgData": "./dsp/assets/svgs/HeadingColors.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "size": {
      "defaultValue": null,
      "description": "The size and type of Heading to be used.\nDenotes hierarchy.",
      "name": "size",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"heading-1\" | \"heading-2\" | \"heading-3\" | \"heading-4\" | \"heading-5\" | \"heading-6\" | undefined"
      }
    },
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI color to be applied to the Heading.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
      }
    },
    "as": {
      "defaultValue": null,
      "description": "",
      "name": "as",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    },
    "m": {
      "defaultValue": null,
      "description": "",
      "name": "m",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "my": {
      "defaultValue": null,
      "description": "",
      "name": "my",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mx": {
      "defaultValue": null,
      "description": "",
      "name": "mx",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mt": {
      "defaultValue": null,
      "description": "",
      "name": "mt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mb": {
      "defaultValue": null,
      "description": "",
      "name": "mb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "ml": {
      "defaultValue": null,
      "description": "",
      "name": "ml",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mr": {
      "defaultValue": null,
      "description": "",
      "name": "mr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "p": {
      "defaultValue": null,
      "description": "",
      "name": "p",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "py": {
      "defaultValue": null,
      "description": "",
      "name": "py",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "px": {
      "defaultValue": null,
      "description": "",
      "name": "px",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pt": {
      "defaultValue": null,
      "description": "",
      "name": "pt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pb": {
      "defaultValue": null,
      "description": "",
      "name": "pb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pl": {
      "defaultValue": null,
      "description": "",
      "name": "pl",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pr": {
      "defaultValue": null,
      "description": "",
      "name": "pr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Heading is used to display headlines and other forms of hierarchical Text.\n\nHeadings are similar to the base Text component, but restricted to certain\nsizes and font weight.",
    "displayName": "Heading",
    "methods": [],
    "props": {
      "size": {
        "defaultValue": null,
        "description": "The size and type of Heading to be used.\nDenotes hierarchy.",
        "name": "size",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"heading-1\" | \"heading-2\" | \"heading-3\" | \"heading-4\" | \"heading-5\" | \"heading-6\" | undefined"
        }
      },
      "color": {
        "defaultValue": null,
        "description": "The Dracula UI color to be applied to the Heading.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
        }
      },
      "as": {
        "defaultValue": null,
        "description": "",
        "name": "as",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      },
      "m": {
        "defaultValue": null,
        "description": "",
        "name": "m",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "my": {
        "defaultValue": null,
        "description": "",
        "name": "my",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mx": {
        "defaultValue": null,
        "description": "",
        "name": "mx",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mt": {
        "defaultValue": null,
        "description": "",
        "name": "mt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mb": {
        "defaultValue": null,
        "description": "",
        "name": "mb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "ml": {
        "defaultValue": null,
        "description": "",
        "name": "ml",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mr": {
        "defaultValue": null,
        "description": "",
        "name": "mr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "p": {
        "defaultValue": null,
        "description": "",
        "name": "p",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "py": {
        "defaultValue": null,
        "description": "",
        "name": "py",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "px": {
        "defaultValue": null,
        "description": "",
        "name": "px",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pt": {
        "defaultValue": null,
        "description": "",
        "name": "pt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pb": {
        "defaultValue": null,
        "description": "",
        "name": "pb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pl": {
        "defaultValue": null,
        "description": "",
        "name": "pl",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pr": {
        "defaultValue": null,
        "description": "",
        "name": "pr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      }
    }
  }
}
