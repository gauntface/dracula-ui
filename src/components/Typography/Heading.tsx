import React from 'react'
import { mapValues } from 'lodash'
import { colors } from '../Color/Box'

export const HeadingSizes = {
  'heading-1': 'drac-heading-1',
  'heading-2': 'drac-heading-2',
  'heading-3': 'drac-heading-3',
  'heading-4': 'drac-heading-4'
}

export const HeadingColors = mapValues(colors, className => {
  return className.replace('-bg-', '-text-')
})

export interface HeadingProps {
  size?: keyof typeof HeadingSizes
  color?: keyof typeof HeadingColors
}

export const Heading: React.FC<HeadingProps> = props => {
  const tag = {
    'heading-1': 'h1',
    'heading-2': 'h2',
    'heading-3': 'h3',
    'heading-4': 'h4'
  }

  const size = tag[props.size ?? 'heading-1']
  const finalProps = {
    ...props,
    className: `
      drac-heading
      ${HeadingSizes[props.size ?? 'heading-1']}
      ${HeadingColors[props.color ?? 'black']}
    `.trim()
  }

  return React.createElement(size, finalProps, props.children)
}
