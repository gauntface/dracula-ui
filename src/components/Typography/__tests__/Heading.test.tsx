import React from 'react'
import { Heading } from '../Heading'
import testRenderer from 'react-test-renderer'

describe('Heading', () => {
  it('heading-1', () => {
    const rendered = testRenderer.create(
      <Heading size="heading-1">Hello World</Heading>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('heading-2', () => {
    const rendered = testRenderer.create(
      <Heading size="heading-2">Hello World</Heading>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('heading-3', () => {
    const rendered = testRenderer.create(
      <Heading size="heading-3">Hello World</Heading>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })

  it('heading-4', () => {
    const rendered = testRenderer.create(
      <Heading size="heading-4">Hello World</Heading>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
