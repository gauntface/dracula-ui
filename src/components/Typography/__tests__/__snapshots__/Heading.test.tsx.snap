// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading heading-1 1`] = `
<h1
  className="drac-heading drac-heading-1 drac-spacing-none"
  size="heading-1"
>
  Hello World
</h1>
`;

exports[`Heading heading-2 1`] = `
<h2
  className="drac-heading drac-heading-2 drac-spacing-none"
  size="heading-2"
>
  Hello World
</h2>
`;

exports[`Heading heading-3 1`] = `
<h3
  className="drac-heading drac-heading-3 drac-spacing-none"
  size="heading-3"
>
  Hello World
</h3>
`;

exports[`Heading heading-4 1`] = `
<h4
  className="drac-heading drac-heading-4 drac-spacing-none"
  size="heading-4"
>
  Hello World
</h4>
`;
